app/client/src/utils/AnalyticsUtil.tsx

```
@@ -393,6 +393,71 @@ class AnalyticsUtil {
+    n.type = "text/partytown";
+    n.innerHTML = `(function init(window) {
+      const analytics = (window.analytics = window.analytics || []);
+      if (!analytics.initialize) {
+        if (analytics.invoked) {
+          console.error("Segment snippet included twice.");
+        } else {
+          analytics.invoked = !0;
+          analytics.methods = [
+            "trackSubmit",
+            "trackClick",
+            "trackLink",
+            "trackForm",
+            "pageview",
+            "identify",
+            "reset",
+            "group",
+            "track",
+            "ready",
+            "alias",
+            "debug",
+            "page",
+            "once",
+            "off",
+            "on",
+          ];
+          analytics.factory = function(t) {
+            return function() {
+              const e = Array.prototype.slice.call(arguments); //eslint-disable-line prefer-rest-params
+              e.unshift(t);
+              analytics.push(e);
+              return analytics;
+            };
+          };
+        }
+        for (let t = 0; t < analytics.methods.length; t++) {
+          const e = analytics.methods[t];
+          analytics[e] = analytics.factory(e);
+        }
+        analytics.load = function(t, e) {
+          const n = document.createElement("script");
+          n.type = "text/partytown";
+          n.async = !0;
+          // Ref: https://www.notion.so/appsmith/530051a2083040b5bcec15a46121aea3
+          n.src = "https://a.appsmith.com/reroute/" + t + "/main.js";
+          const a = document.getElementsByTagName("script")[0];
+          a.parentNode.insertBefore(n, a);
+          analytics._loadOptions = e;
+        };
+        analytics.SNIPPET_VERSION = "4.15.3";
+        analytics.load("${key}");
+        analytics.page();
+      }
+    })(window);`;
+    try {
+      const a: any = document.getElementsByTagName("script")[0];
+      a.parentNode.insertBefore(n, a);
+      return Promise.resolve(true);
```

This seems to undo the changes we did for handling `anonymousId` for on page load actions [here](https://github.com/appsmithorg/appsmith/pull/19122/files#diff-ed85344d831d6f31c8587a8155004d9a511debc75e3ec69f894919d8ade6dd89:~:text=%7D%3B-,analytics.ready,-(())).

We want the Promise to be resolved with true, only when analytics.ready is triggered.
And with false, with a maximum wait time of 2 seconds because:
1. We don't have an analytics.error event (at least I couldn't find one)
2. We'd be holding off on page load actions which are user actions. So, we'd rather fail fast instead of waiting on analytics.